### PR TITLE
Done task feature

### PR DESCRIPTION
**Pull request by ahmedzaky030**

### CODE REVIEW

File: src/main/webapp/app/tasks/task-list/task-list.component.html [hunk 1]
```diff
@@ -1,7 +1,13 @@
+<button mat-icon-button aria-label="Delete task" color="primary">
+  Clear All Done Tasks
+  <mat-icon aria-label="Delete task" (click)="clearAllDoneTasks()">delete</mat-icon>
```
Comment (djuerges) [inline src/main/webapp/app/tasks/task-list/task-list.component.html:3]: it is a bit unexpected just to have the action on the icon, not the text too

File: src/main/webapp/app/tasks/task-list/task-list.component.html [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+</button>
 <mat-list data-cy="task-list">
-  <mat-list-item *ngFor="let task of tasks" class="mat-elevation-z1">
+  <mat-list-item *ngFor="let task of tasks; let index = index" class="mat-elevation-z1">
+    <mat-icon *ngIf="task.isDone" mat-list-icon>done</mat-icon>
     <mat-icon mat-list-icon>assignment</mat-icon>
     <h4 mat-line>{{task.name}}</h4>
+    <mat-slide-toggle [checked]="task?.isDone" [color]="'primary'" (change)="checkingDone($event, index, tasks)" > Done</mat-slide-toggle>
```
Comment (djuerges) [inline src/main/webapp/app/tasks/task-list/task-list.component.html:10]: fitting better into the present UI one could have gone with a checkbox too

File: src/main/webapp/app/tasks/task-list/task-list.component.ts [hunk 3]
```diff
@@ -12,17 +13,42 @@ import { TaskService } from '../task.service';
...
+  }
+
+  clearAllDoneTasks() {
+    this.tasks.filter(task => task.isDone).forEach(task => {
+      this.taskService.delete(task.id).subscribe(() => {
+        // nothing
+      });
+      this.deleted.emit();
+    });
+  }
```
Comment (djuerges) [inline src/main/webapp/app/tasks/task-list/task-list.component.ts:53]: would have liked to see some tests for this

File: src/main/webapp/app/tasks/task-list/task-list.component.html (hunk 1 above)
Comment (ahmedzaky030) [reply src/main/webapp/app/tasks/task-list/task-list.component.html]: I moved it to the button tag, you're right. It's unexpected.

File: src/main/webapp/app/tasks/task-list/task-list.component.html (hunk 2 above)
Comment (ahmedzaky030) [reply src/main/webapp/app/tasks/task-list/task-list.component.html]: I've changed it to a checkbox.

File: src/main/webapp/app/tasks/task-list/task-list.component.ts (hunk 3 above)
Comment (ahmedzaky030) [reply src/main/webapp/app/tasks/task-list/task-list.component.ts:52]: I've written tests for this and refactor the code to be testable.

**Pull request closed without merging**